### DISCUSSION TITLE
Failed to proxy/rewrite self-signed https url

### DISCUSSION THREAD

**Discussion by aslamj** [vercel/next.js#29947] [Help]
What I wonder is that if I'm writing an API using NextJS rules and any handler method where I do a fetch to another API with a non trusted CA cert (may be self-signed one), then having an environment variable ERR_CERT_AUTHORITY_INVALID=0, skips the cert checking totally. So why not use the same for the rewrite runtime code and make it work for us.

In lower environments (dev and test), normally any internal dependent API server (where API traffic is being routed to) will have a non-trusted server cert or a non-trusted self-signed-cert. And this these lower environment, the rewrite code get a cert error like: UNABLE_TO_GET_ISSUER_CERT_LOCALLY

** @type {import('next').NextConfig} */
module.exports = {
  reactStrictMode: true,
  async rewrites() {
    return [
      {
        source: '/api/v1/:path*',
        destination: 'https://self-signed-cert:443/api/v1/:path*',
      },
    ]
  },
}

_Originally posted by @aslamj in https://github.com/vercel/next.js/issues/21537#issuecomment-944021424_

**Comment by zomars** [comment]
I'm having this same issue,

**Comment by KurtGokhan** [comment]
Can we think about this request? I see it was discussed in #21537 and has got some upvotes, but closed because it was ~~inactive~~ ignored.

**Reply by jayhill90** [reply]
seriously tho

**Comment by alexcybernetic** [comment]
Why nextjs is still not respecting NODE_TLS_REJECT_UNAUTHORIZED=0 in a dev environment?
Or is it something different?

Next.js 14.1.0
Node.js v21.6.1

![Screenshot 2024-01-31 at 17 00 09](https://github.com/vercel/next.js/assets/7684726/ee12affa-c4cb-434f-a3e0-0ba14d470abd)

**Comment by jmotes** [comment]
I found a solution for my local development docker-compose based setup.

I wrote a script to copy my Caddy container's root cert into my host machine and then copy it into all the other containers. This allowed me to specify the path to the cert via the CLI along with `npm run dev`. 

This works even though Next.js doesn't seem to read `NODE_EXTRA_CA_CERTS` from my `.env` file!

```
NODE_EXTRA_CA_CERTS=/usr/local/share/ca-certificates/root.crt npm run dev
```

Feel free to ask if you need any guidance on the specific commands to copy the files from Caddy to the other containers.
